tests/output.rs

```diff
@@ -1,228 +1,118 @@
+fn do_sorted_lines_match(actual_output: &str, expected_output: &str) -> bool {
+    let mut actual_lines = actual_output.split("\n").collect::<Vec<_>>();
+    actual_lines.sort();
+    let mut expected_lines = expected_output.split("\n").collect::<Vec<_>>();
+    expected_lines.sort();
+    actual_lines == expected_lines
 }
 
-impl FixtureQueryExpectedMatches {
-    pub fn assert_expected_matches(&self, output_mode: OutputMode) {
-        let mut args = vec![
-            "--query-source",
-            &self.query_source,
-            "--language",
-            &self.language,
-        ];
-        if output_mode == OutputMode::Vimgrep {
-            args.push("--vimgrep");
-        }
-        assert_expected_matches(
-            args,
-            get_fixture_dir_path_from_name(&self.fixture_dir_name),
-            &self.expected_matches,
-            output_mode,
-        );
-    }
-}
-
-fn get_fixture_dir_path_from_name(fixture_dir_name: &str) -> PathBuf {
-    // per https://andrewra.dev/2019/03/01/testing-in-rust-temporary-files/
-    let root_dir = env::var("CARGO_MANIFEST_DIR").unwrap();
-    let mut path: PathBuf = root_dir.into();
-    path.push("tests/fixtures");
-    path.push(fixture_dir_name);
-    path
-}
-
-const FUNCTION_ITEM_QUERY_SOURCE: &'static str = "(function_item) @function_item";
-
-static RUST_PROJECT_FUNCTION_ITEM_EXPECTED_MATCHES: Lazy<FixtureQueryExpectedMatches> =
-    Lazy::new(|| FixtureQueryExpectedMatches {
-        fixture_dir_name: "rust_project".into(),
-        query_source: FUNCTION_ITEM_QUERY_SOURCE.to_owned(),
-        language: "rust".to_owned(),
-        expected_matches: vec![
-            ExpectedMatch {
-                relative_file_path: "src/helpers.rs".to_owned(),
-                lines: vec!["pub fn helper() {}".to_owned()],
-                range: ((0, 0), (0, 18)).into(),
-            },
-            ExpectedMatch {
-                relative_file_path: "src/lib.rs".to_owned(),
-                lines: vec![
-                    "pub fn add(left: usize, right: usize) -> usize {".to_owned(),
-                    "    left + right".to_owned(),
-                    "}".to_owned(),
-                ],
-                range: ((2, 0), (5, 0)).into(),
-            },
-            ExpectedMatch {
-                relative_file_path: "src/lib.rs".to_owned(),
-                lines: vec![
-                    "    fn it_works() {".to_owned(),
-                    "        let result = add(2, 2);".to_owned(),
-                    "        assert_eq!(result, 4);".to_owned(),
-                    "    }".to_owned(),
-                ],
-                range: ((11, 4), (14, 4)).into(),
-            },
-        ],
-    });
-
 #[test]
 fn test_query_inline() {
-    RUST_PROJECT_FUNCTION_ITEM_EXPECTED_MATCHES.assert_expected_matches(OutputMode::Normal);
+    assert_sorted_output(
+        "rust_project",
+        r#"
+            $ tree-sitter-grep --query-source '(function_item) @function_item' --language rust
```

Comment:
I think this is great, the pattern I'd set up was feeling clunky/not exactly paying huge dividends for "reuse" so figured I'd switch to this